### PR TITLE
add case verify_enable_kprobe

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

File: microsoft/testsuites/core/debug.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+@TestSuiteMetadata(
+    area="core",
+    category="functional",
+    description="""
+    This test suite covers kernel debug functionalities.
+    """,
+    requirement=simple_requirement(unsupported_os=[]),
+)
+class Debug(TestSuite):
```
Comment (squirrelsc) [inline microsoft/testsuites/core/debug.py]: `KernelDebug`

File: lisa/tools/mount.py [hunk 2]
```diff
@@ -150,6 +153,19 @@ def get_mount_point_for_partition(self, partition_name: str) -> Optional[str]:
...
+        output: str = self.run(force_run=True).stdout
+        mount_points: List[str] = []
+        for line in output.splitlines():
+            matched = self._mount_info_regex.fullmatch(line)
+            if matched:
+                mount_points.append(
+                    matched.group("mount_point"),
+                )
+
+        self._log.debug(f"Found all mount points : {mount_points}")
```
Comment (squirrelsc) [inline lisa/tools/mount.py]: "Found mount points: "

File: lisa/tools/mount.py [hunk 3]
```diff
@@ -150,6 +153,19 @@ def get_mount_point_for_partition(self, partition_name: str) -> Optional[str]:
 
         return partition.mount_point
 
+    def check_mount_point_exist(self, mount_point: str) -> bool:
+        output: str = self.run(force_run=True).stdout
+        mount_points: List[str] = []
+        for line in output.splitlines():
```
Comment (squirrelsc) [inline lisa/tools/mount.py]: Use `util.find_groups_in_lines`

File: microsoft/testsuites/core/debug.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+                    )
+                ).is_equal_to("p:kprobes/my filp_close")
+                assert_that(
+                    cat.read(
+                        "/sys/kernel/debug/tracing/events/kprobes/my/enable",
+                        sudo=True,
+                        force_run=True,
+                    )
+                ).is_equal_to("1")
+            except Exception as ex:
```
Comment (squirrelsc) [inline microsoft/testsuites/core/debug.py]: It do nothing, so can be totally removed. `try... finally...` is enough.

File: microsoft/testsuites/core/debug.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+                cat = node.tools[Cat]
+                assert_that(
+                    cat.read(
+                        "/sys/kernel/debug/tracing/kprobe_events",
+                        sudo=True,
+                        force_run=True,
+                    )
+                ).is_equal_to("p:kprobes/my filp_close")
+                assert_that(
```
Comment (squirrelsc) [inline microsoft/testsuites/core/debug.py]: add describe for each assertion.

File: microsoft/testsuites/core/debug.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+                    ),
+                    sudo=True,
+                )
+                echo.write_to_file(
+                    origin_kprobe_events,
+                    node.get_pure_path("/sys/kernel/debug/tracing/kprobe_events"),
+                    sudo=True,
+                )
+        else:
+            raise SkippedException("CONFIG_KPROBE_TRACING is not enabled")
```
Comment (squirrelsc) [inline microsoft/testsuites/core/debug.py]: Raise exception at the beginning, so it's not a long `if...else`. It helps to understand logic easier.

File: lisa/tools/mount.py (hunk 2 above)
Comment (LiliDeng) [reply lisa/tools/mount.py]: updated

File: microsoft/testsuites/core/debug.py (hunk 4 above)
Comment (LiliDeng) [reply microsoft/testsuites/core/debug.py]: updated

File: microsoft/testsuites/core/debug.py (hunk 5 above)
Comment (LiliDeng) [reply microsoft/testsuites/core/debug.py]: updated

File: microsoft/testsuites/core/debug.py (hunk 6 above)
Comment (LiliDeng) [reply microsoft/testsuites/core/debug.py]: updated

File: microsoft/testsuites/core/debug.py [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+         /sys/kernel/debug/tracing/events/kprobes/my/enable.
+
+        """,
+        priority=1,
+        requirement=simple_requirement(supported_platform_type=[AZURE, READY]),
+    )
+    def verify_enable_kprobe(self, node: Node) -> None:
+        if not node.tools[KernelConfig].is_enabled("CONFIG_KPROBE_EVENTS"):
+            raise SkippedException("CONFIG_KPROBE_TRACING is not enabled")
+        else:
```
Comment (squirrelsc) [inline microsoft/testsuites/core/debug.py]: The `else` is not necessary. When the `if` block is true, and exception happens, the following code won't be executed. Removing the `else` will reduce code complex.

Comment (LiliDeng) [reply microsoft/testsuites/core/debug.py]: updated

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**